lib/rubocop/cop/rspec/subject_stub.rb

```diff
@@ -73,72 +77,38 @@ class SubjectStub < Cop
             } ...)
         PATTERN
 
-        def on_block(node)
-          return unless example_group?(node)
+        def on_top_level_describe(node, _args)
+          @explicit_subjects = find_all_explicit_subjects(node.parent)
 
-          find_subject_stub(node) do |stub|
+          find_subject_expectations(node.parent) do |stub|
             add_offense(stub)
           end
         end
 
         private
 
-        # Find subjects within tree and then find (send) nodes for that subject
-        #
-        # @param node [RuboCop::Node] example group
-        #
-        # @yield [RuboCop::Node] message expectations for subject
-        def find_subject_stub(node, &block)
-          find_subject(node) do |subject_name, context|
-            find_subject_expectation(context, subject_name, &block)
+        def find_all_explicit_subjects(node)
+          node.each_descendant(:block).each_with_object({}) do |child, h|
+            name = subject(child)
+            if name
+              h[child.parent.parent] ||= []
```

Comment:
what's the meaning of `parent.parent`? isn't it node.parent? while you are attaching to the parent?

Reply:
parent should be a `block` node for outer `describe`/`context`/...

Reply:
Will it work properly for contexts defined in iterators? (yes, another funny concept from my current project)
```ruby
RSpec.describe do
  %i[one two].each do |count|
    context "for #{count}" do
      specify do
        expect(subject).to receive(count).and_return(0)
```
There's a parent block, but not an example group block.
I'm not sure this is the example to fail, hope you can figure out a better example to fail `.parent.parent`.